site/zenodo_rdm/legacy/requests/record_upgrade.py

```diff
@@ -0,0 +1,182 @@
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Zenodo legacy record upgrade request."""
+
+from invenio_access.permissions import system_identity
+from invenio_i18n import gettext as _
+from invenio_rdm_records.proxies import (
+    current_rdm_records_service,
+    current_record_communities_service,
+)
+from invenio_records_resources.services.uow import IndexRefreshOp, RecordCommitOp
+from invenio_requests.customizations import CommentEventType, RequestType, actions
+from invenio_requests.proxies import current_events_service, current_requests_service
+
+
+def _remove_permission_flag(record, uow):
+    """Remove can_community_read_files permission flag."""
+    try:
+        record.parent.permission_flags.pop("can_community_read_files")
+        record.parent.commit()
+        uow.register(RecordCommitOp(record.parent))
+    except (KeyError, AttributeError):
+        # if field or flag is absent just continue with an action
+        pass
+
+
+def _remove_record_from_communities(record, identity):
+    """Remove record from all the communities."""
+    communities_ids = record.parent.communities.ids
+    communities = []
+    for community_id in communities_ids:
+        communities.append({"id": community_id})
+
+    data = dict(communities=communities)
+    current_record_communities_service.remove(identity, record["id"], data)
+
+
+def _create_comment(request, content, uow):
+    """Create a comment event."""
+    comment = {"payload": {"content": _(content)}}
+
+    current_events_service.create(
+        system_identity, request.id, comment, CommentEventType, uow=uow
+    )
+
+    # make event immediately available in search
+    uow.register(IndexRefreshOp(indexer=current_events_service.indexer))
+
+
+#
+# Actions
+#
+class SubmitAction(actions.SubmitAction):
+    """Submit action."""
+
+    def execute(self, identity, uow):
+        """Execute the submit action."""
+        record = self.request.topic.resolve()
+        # Set the record's title as the request title.
+        self.request["title"] = record.metadata["title"]
+
+        # example: "May 11, 2024"
+        expires_at = self.request.expires_at.strftime("%B %d, %Y")
+        self.request["description"] = _(
+            "<b>Upgrade legacy record request</b><br><br>"
+            "The current record uses legacy behaviours which is being removed from "
+            "the platform and is in need of being upgraded. Upgrading the record "
+            "will have the following impact:"
+            "<ul><li><b>Access: </b>curators of communities that this record "
+            "is part of will be granted view and edit access to your record (i.e. curators can see restricted "
+            "files and edit metadata)</li></ul>"
+            f"If the record has not been upgraded by <b>{expires_at}</b>, it will"
+            " be <u>removed from the communities</u> it is part of and upgraded."
+        )
```

Comment:
```suggestion
        self.request["description"] = (
            "The current record uses legacy behavior which has been removed from "
            "the platform, and needs to be upgraded. Upgrading the record "
            "will have the following impact:"
            "<ul><li><b>Access: </b>curators of communities that this record "
            "is part of will be granted view and edit access to your record (i.e. curators can see restricted "
            "files and edit metadata)</li></ul>"
            f"If you do not upgrade the record by <b>{expires_at}</b>, it will "
            "be automatically <u>removed from its communities</u>."
        )
```

- Removed i18n (since we're not using translations in Zenodo)
- Typos/phrasing
- Changed wording from passive to active (to make it clear that the user has to do something :) )